### PR TITLE
Add glide.yaml  and vendor folder to top level

### PR DESCRIPTION
**Pull request by wgottschalk**
resolves https://github.com/Originate/exosphere-sdk/issues/285
WIP 
the plan is to aggregate all of the `glide.yaml` files to a top level file. Then delete the glide files for each subdirectory.
Not sure what to do with the package statements though. Does it change to just 
`github.com/Originate/exocom/go`?

### CODE REVIEW

File: go/glide.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package: github.com/Originate/exocom/go
+import:
+- package: github.com/fatih/color
+  version: v1.1.0
+- package: golang.org/x/net
+  subpackages:
+  - websocket
+- package: gopkg.in/tylerb/graceful.v1
+  version: v1.2.13
```
Comment (charlierudolph) [inline go/glide.yaml]: `github.com/fatih/color` and `gopkg.in/tylerb/graceful.v1` are unused and should be removed.

**Comment by charlierudolph** [comment]
I think the `bin/setup` command in each folder should be combined into a `bin/setup` in the go directory. The top level `bin/setup_ci` will need to be updated to just call that instead of calling instead of `../morula all bin/setup`

**Comment by wgottschalk** [comment]
@charlierudolph @kevgo ready for final review. Lots of files changed due to moving vendor folders but the important ones should be at the top

File: go/bin/lint [hunk 2]
```diff
@@ -3,5 +3,5 @@ set -e
 
 # required by gotype to work properly
 # see https://github.com/golang/go/issues/12703#issuecomment-144429128
-go install
-gometalinter.v1 --deadline 60s ./ ./test-fixtures
+go install $(go list ./... | grep -v /vendor/)
+gometalinter.v1 --deadline 60s ./
```
Comment (charlierudolph) [inline go/bin/lint]: gometalinter is not recursive so you need to specify the directories you want to lint

File: go/bin/spec [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -e
+
+bin/lint
+
+# runs go test on all subdirectories exluding the vendor folder
+# might cause issues with windows users though
```
Comment (charlierudolph) [inline go/bin/spec]: Why might this cause issues with windows users?

File: go/exorelay/exorelay_test.go [hunk 4]
```diff
@@ -73,6 +74,7 @@ func FeatureContext(s *godog.Suite) {
 	})
 
 	s.Step(`^ExoRelay connects to Exocom$`, func() error {
+		time.Sleep(time.Millisecond * 1)
```
Comment (charlierudolph) [inline go/exorelay/exorelay_test.go]: Thoughts on making connect retry on failure? (with a timeout of 1 second)

File: go/bin/spec (hunk 3 above)
Comment (wgottschalk) [inline go/bin/spec]: can windows users run bash commands?

File: go/exorelay/exorelay_test.go (hunk 4 above)
Comment (wgottschalk) [inline go/exorelay/exorelay_test.go]: I think it's not in the scope of this PR but it's something that we should be doing

Comment (charlierudolph) [inline go/exorelay/exorelay_test.go]: K. Created an issue for it https://github.com/Originate/exosphere-sdk/issues/302

File: go/bin/spec (hunk 3 above)
Comment (charlierudolph) [inline go/bin/spec]: http://gnuwin32.sourceforge.net/packages/grep.htm
How about updating the comment to: `(windows users need to 'grep' installed)`

Comment (charlierudolph) [inline go/bin/spec]: Actually thoughts on moving all our code into `go/src/` so that we don't need to filter out the vendor folder?

Comment (wgottschalk) [inline go/bin/spec]: I was actually looking at morula and it looks like there is a never command to exclude certain directories. So I could just say `cd go && ../morula all --never=vendor bin/<command>` in the top bin folder.

Comment (charlierudolph) [inline go/bin/spec]: That sounds good. Will also need to exclude the `bin` folder if we are keeping the shared setup

**Pull request closed without merging**